api/rest/client/methods.go

```diff
@@ -172,3 +178,106 @@ func (c *Client) GetConnectGraph() (api.ConnectGraphSerial, error) {
 	err := c.do("GET", "/health/graph", nil, &graphS)
 	return graphS, err
 }
+
+type StatusFilter struct {
+	In, Out chan api.GlobalPinInfo
+	Done    chan struct{}
+	Err     chan error
+}
+
+func NewStatusFilter() *StatusFilter {
+	return &StatusFilter{
+		In:   make(chan api.GlobalPinInfo),
+		Out:  make(chan api.GlobalPinInfo),
+		Done: make(chan struct{}),
+		Err:  make(chan error),
+	}
+}
+
+type FilterParams struct {
+	*cid.Cid
+	Local     bool
+	Target    api.TrackerStatus
+	CheckFreq time.Duration
+}
+
+func (sf *StatusFilter) Filter(ctx context.Context, fp FilterParams) {
+	go func() {
+		for {
+			select {
+			case <-ctx.Done():
+				sf.Err <- ctx.Err()
```

Comment:
If channels are exported they should be closed when the sender is not going to send anything anymore. Whoever uses them may just be hanging. I am leaning towards not exporting any channel (nor the `StatusFilter`), and just have `WaitFor` provide the functionality we need (it is easy to reverse this decision later if needed).